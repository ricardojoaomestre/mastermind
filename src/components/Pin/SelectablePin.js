import styled from "styled-components";
import Pin from "./Pin";
import { IoCheckmarkSharp } from "react-icons/io5";

const PinIcon = styled(IoCheckmarkSharp)`
  width: 20px;
  height: 20px;
  stroke: #ffffff;
`;

const SelectablePin = ({ selected, ...otherProps }) => (
  <Pin {...otherProps}>{selected && <PinIcon />}</Pin>
);

export default SelectablePin;
