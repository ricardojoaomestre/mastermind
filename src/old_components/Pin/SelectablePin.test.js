import { render, cleanup } from '@testing-library/react';

import SelectablePin from './SelectablePin';
import { IoCheckmarkOutline } from 'react-icons/io5';
afterEach(cleanup);

describe('SelectablePin', () => {
    it("must render as expected", () => {
        const { container } = render(<SelectablePin />);
        expect(container.firstChild).toMatchSnapshot();
    });

    xit("must display an icon when selected", () => {   
    });
});