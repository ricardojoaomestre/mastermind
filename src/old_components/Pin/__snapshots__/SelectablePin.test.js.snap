// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectablePin must render as expected 1`] = `
<div
  class="sc-bdfBQB sc-gsTEea ioqCIK dCYNna"
/>
`;
